TestUnpackMalformedBytes fails on 32 bit architectures

`TestUnpackMalformedBytes` was introduced in https://github.com/google/go-tpm/commit/16c6370942f77e02eb029c5f3ee5c7ba610eece0. I'm seeing failures when building on i686 or armv7hl:
```
github.com/google/go-tpm/tpmutil
--- FAIL: TestUnpackMalformedBytes (0.00s)
panic: runtime error: slice bounds out of range [:-1] [recovered]
	panic: runtime error: slice bounds out of range [:-1]
goroutine 24 [running]:
testing.tRunner.func1.2(0x567807c0, 0x588187d0)
	/usr/lib/golang/src/testing/testing.go:1143 +0x2ce
testing.tRunner.func1(0x5891a460)
	/usr/lib/golang/src/testing/testing.go:1146 +0x3fc
panic(0x567807c0, 0x588187d0)
	/usr/lib/golang/src/runtime/panic.go:971 +0x491
github.com/google/go-tpm/tpmutil.(*U32Bytes).TPMUnmarshal(0x5880e2e0, 0x5678e3a8, 0x58812498, 0xe623f048, 0x5880e2e0)
	/builddir/build/BUILD/go-tpm-0.3.2/_build/src/github.com/google/go-tpm/tpmutil/structures.go:113 +0x244
github.com/google/go-tpm/tpmutil.tryUnmarshal(0x5678e3a8, 0x58812498, 0x56775160, 0x5880e2e0, 0x16, 0x0, 0x56632155, 0x5662bb94)
	/builddir/build/BUILD/go-tpm-0.3.2/_build/src/github.com/google/go-tpm/tpmutil/encoding.go:138 +0xf2
github.com/google/go-tpm/tpmutil.unpackValue(0x5678e3a8, 0x58812498, 0x56775160, 0x5880e2e0, 0x16, 0x568348e0, 0xf7cfd1c8)
	/builddir/build/BUILD/go-tpm-0.3.2/_build/src/github.com/google/go-tpm/tpmutil/encoding.go:164 +0x8c
github.com/google/go-tpm/tpmutil.UnpackBuf(0x5678e3a8, 0x58812498, 0x5884af48, 0x1, 0x1, 0x567706e0, 0x2106601)
	/builddir/build/BUILD/go-tpm-0.3.2/_build/src/github.com/google/go-tpm/tpmutil/encoding.go:206 +0x109
github.com/google/go-tpm/tpmutil.Unpack(0x588187c0, 0x5, 0x5, 0x58838748, 0x1, 0x1, 0x55acc, 0x1, 0x21066bb)
	/builddir/build/BUILD/go-tpm-0.3.2/_build/src/github.com/google/go-tpm/tpmutil/encoding.go:155 +0x89
github.com/google/go-tpm/tpmutil.TestUnpackMalformedBytes(0x5891a460)
	/builddir/build/BUILD/go-tpm-0.3.2/_build/src/github.com/google/go-tpm/tpmutil/encoding_test.go:361 +0xad
testing.tRunner(0x5891a460, 0x5678cd10)
	/usr/lib/golang/src/testing/testing.go:1193 +0x102
created by testing.(*T).Run
	/usr/lib/golang/src/testing/testing.go:1238 +0x233
exit status 2
FAIL	github.com/google/go-tpm/tpmutil	0.008s
```
You can see the full logs here at https://koji.fedoraproject.org/koji/taskinfo?taskID=70436942 and https://koji.fedoraproject.org/koji/taskinfo?taskID=70436941 (look at build.log).

--- a/tpmutil/structures.go
+++ b/tpmutil/structures.go
@@ -26,7 +26,7 @@ import (
 // from resulting in a massive memory allocation, potentially causing
 // an OOM condition on the system.
 // We expect no buffer from a TPM to approach 1Mb in size.
-const maxBytesBufferSize = 1024 * 1024 // 1Mb.
+const maxBytesBufferSize uint32 = 1024 * 1024 // 1Mb.
 
 // RawBytes is for Pack and RunCommand arguments that are already encoded.
 // Compared to []byte, RawBytes will not be prepended with slice length during
@@ -38,8 +38,8 @@ type U16Bytes []byte
 
 // TPMMarshal packs U16Bytes
 func (b *U16Bytes) TPMMarshal(out io.Writer) error {
-	size := uint16(len([]byte(*b)))
-	if err := binary.Write(out, binary.BigEndian, size); err != nil {
+	size := len([]byte(*b))
+	if err := binary.Write(out, binary.BigEndian, uint16(size)); err != nil {
 		return err
 	}
 
@@ -47,7 +47,7 @@ func (b *U16Bytes) TPMMarshal(out io.Writer) error {
 	if err != nil {
 		return err
 	}
-	if n != int(size) {
+	if n != size {
 		return fmt.Errorf("unable to write all contents of U16Bytes")
 	}
 	return nil
@@ -82,8 +82,8 @@ type U32Bytes []byte
 
 // TPMMarshal packs U32Bytes
 func (b *U32Bytes) TPMMarshal(out io.Writer) error {
-	size := uint32(len([]byte(*b)))
-	if err := binary.Write(out, binary.BigEndian, size); err != nil {
+	size := len([]byte(*b))
+	if err := binary.Write(out, binary.BigEndian, uint32(size)); err != nil {
 		return err
 	}
 
@@ -91,7 +91,7 @@ func (b *U32Bytes) TPMMarshal(out io.Writer) error {
 	if err != nil {
 		return err
 	}
-	if n != int(size) {
+	if n != size {
 		return fmt.Errorf("unable to write all contents of U32Bytes")
 	}
 	return nil
@@ -103,11 +103,12 @@ func (b *U32Bytes) TPMUnmarshal(in io.Reader) error {
 	if err := binary.Read(in, binary.BigEndian, &tmpSize); err != nil {
 		return err
 	}
-	size := int(tmpSize)
 
-	if size > maxBytesBufferSize {
+	if tmpSize > maxBytesBufferSize {
 		return bytes.ErrTooLarge
 	}
+	// We can now safely cast to an int on 32-bit or 64-bit machines
+	size := int(tmpSize)
 
 	if len(*b) >= size {
 		*b = (*b)[:size]

